channeldb/payments.go

```diff
@@ -570,7 +578,21 @@ type PaymentsResponse struct {
 // to a subset of payments by the payments query, containing an offset
 // index and a maximum number of returned payments.
 func (d *DB) QueryPayments(query PaymentsQuery) (PaymentsResponse, error) {
-	var resp PaymentsResponse
+	var (
+		resp           PaymentsResponse
+		minCreationSet = !query.MinCreationTimeNs.IsZero()
+		maxCreationSet = !query.MaxCreationTimeNs.IsZero()
+	)
+
+	// If MinCreationTimeNs filter is greater than MaxCreationTimeNs, we
+	// know there will be no matching results and can safely return early.
+	if minCreationSet && maxCreationSet {
+		if query.MinCreationTimeNs.After(query.MaxCreationTimeNs) {
+			return resp, fmt.Errorf(
+				"MinCreationTimeNs is greater " +
```

Comment:
Nit: I'd include the values in this error so that they're bubbled up to the user for easier debudding when you provide the wrong vals.